source/plugins/ruby/kubelet_utils.rb

```diff
@@ -56,9 +56,9 @@ def get_node_allocatable(cpu_capacity, memory_capacity)
         parsed_response = JSON.parse(allocatable_response.body)
 
         begin
-          kubereserved_cpu = parsed_response["kubeletconfig"]["kubeReserved"]["cpu"]
-          if kubereserved_cpu.nil? || kubereserved_cpu == ""
-            kubereserved_cpu = "0.0"
+          kubereserved_cpu = "0.0"
+          if parsed_response.key?("kubeletconfig") && parsed_response["kubeletconfig"].key?("kubeReserved") && parsed_response["kubeletconfig"]["kubeReserved"].key?("cpu")
```

Comment:
You haven't added the empty string check: kubereserved_cpu == "". The keys can be present but the value can still be empty right?

Same everywhere else

Reply:
from the previous logic, if the kubereserved_cpu == "" then the value will be overriding to 0.

Reply:
addressed